db/question.php

```diff
@@ -60,3 +60,27 @@ function questionViewAdd($question_id) {
       echo "dbの実行に失敗しました。管理者への連絡をお願いします。";
     }
   }
+  
+  // 質問検索用
+  function searchQuestionWithWord($search_word) {
+    include dirname(__FILE__).'/userConnetcion.php';
+
+    try {
+      //sql文作成
+      $sql = "select q.question_id,question_title,question_detail,question_created,question_bestanswer,COUNT(ans.answer_id) as answer_count FROM question as q
+      left outer join answer as ans on q.question_id = ans.question_id
+      where q.question_title like :keyword
+      group by q.question_id;";
```

Comment:
SELECT で指定しているカラムはすべて GROUP BY でも指定しましょう。
今回使っているMySQLではたまたま動いていますが、これを許容しないDBMSもあります。

```suggestion
      //sql文作成
      $sql = "select q.question_id,question_title,question_detail,question_created,question_bestanswer,COUNT(ans.answer_id) as answer_count FROM question as q
      left outer join answer as ans on q.question_id = ans.question_id
      where q.question_title like :keyword
      group by q.question_id, question_title, question_detail, question_created, question_bestanswer;";
```

Reply:
ありがとうございます、修正しました。
https://github.com/2022-team-bulletin-board/2022-bulletin-board-application/pull/14/commits/00ba1aec3f3bd88f5914ce7073f0c7fb4cd77be7

Reply:
2点指摘です。
- 細かいですが、GROUP BY のカラムの方にも `q.` とつけましょう。
- SQL の演算子の優先順位(ANDとOR)について調べてみてください。